### ISSUE TITLE
The Z-level issue

### ISSUE THREAD

**Issue by metalgearsloth** [space-wizards/space-station-14#7925]
I talked with Silicons about the requirements for this which I'm putting below with their permission:
![image](https://user-images.githubusercontent.com/31366439/166618130-7b5313ec-b5b7-4334-8c4a-4cb54b0642aa.png)

So from a design perspective AFAIK it would be:
- Need a way for maps to specify what their zindex is (this should probably be opt-in given the cargo or chess dimensions likely won't need to be considered)
- Need to handle falling in open tiles to lower z-levels (if they exist).
- Need cross-map weldjoints for grids (UGH)
- Might need to just natively have PVS also send chunks for the z-level below the player's current one so rendering and pop-in is just easier.
- Need to somehow also render the z-level below... performantly...
- Need to handle grids moving between z-levels where there may be an existing grid in the way.
- Cross Z-Level power?
- Cross Z-Level grid splitting
- Atmos

Not all of it needs to be done for an initial PR so you can probably get away with supporting traversal first, then physics, then atmos, then working on the rest.

**Comment by Emisse** [comment]
wouldnt z levels and a moving station be fucked up

**Comment by metalgearsloth** [comment]
> wouldnt z levels and a moving station be fucked up

Hence the weldjoint between them. They won't always be synced but it will try its best.

**Comment by moonheart08** [comment]
Would the game eat shit if we simply parented all maps in a group to a multi-z grouping entity.

**Comment by mirrorcult** [comment]
there is absolutely no way maps support being parented to anything right now

**Comment by metalgearsloth** [comment]
Would be better to avoid if possible just due to the extra parent traversing literally everything would have to do.

**Comment by metalgearsloth** [comment]
Yeah a bunch of engine stuff explicitly checks for invalid parent to assume it's the map.

**Comment by moonheart08** [comment]
Also a big open question is how would we save/load these.

**Comment by mirrorcult** [comment]
game map prototypes have to specify the different Z-levels and then the game map loader system loads them all in separate maps and manually welds them together I guess?

**Comment by lvvova1** [comment]
About transporting Cross Z-Level power/pipes

We can make Z-variants of cable/pipes whos installs on wall and moves power/gases btw Z levels

**Comment by ElectroJr** [comment]
Linking here for discoverability: There's currently some multi-z discussion happening [on discord](https://discord.com/channels/1401298992826159275/1423051655049515140). And there was also another older discussion that happened [some time ago](https://discord.com/channels/310555209753690112/1008709214006427689/1074877600431870032).

**Comment by SlamBamActionman** [comment]
Personally not a fan of Z-levels, but I know there is enough interest (especially downstreams) for this to be an active feature request.
